WORKSPACE

```diff
@@ -1,52 +1,79 @@
 workspace(name = "angular_devkit")
 
+# We get Buildifier from here.
+BAZEL_BUILDTOOLS_VERSION = "0.15.0"
+http_archive(
+    name = "com_github_bazelbuild_buildtools",
+    url = "https://github.com/bazelbuild/buildtools/archive/%s.zip" % BAZEL_BUILDTOOLS_VERSION,
+    strip_prefix = "buildtools-%s" % BAZEL_BUILDTOOLS_VERSION,
+    sha256 = "76d1837a86fa6ef5b4a07438f8489f00bfa1b841e5643b618e01232ba884b1fe",
+)
+
+load("@com_github_bazelbuild_buildtools//buildifier:deps.bzl", "buildifier_dependencies")
+buildifier_dependencies()
+
+# The Go toolchain is used for Buildifier.
+# rules_typescript_dependencies() also tries to load it but we add it explicitely so we
+# don't have hidden dependencies. 
+# This also means we need to load it before rules_typescript_dependencies().
+RULES_GO_VERSION = "0.14.0"
+http_archive(
+    name = "io_bazel_rules_go",
+    url = "https://github.com/bazelbuild/rules_go/archive/%s.zip" % RULES_GO_VERSION,
+    strip_prefix = "rules_go-%s" % RULES_GO_VERSION,
+    sha256 = "9bd7c2743f014e4e112b671098ba1da6aec036fe07093b10ca39a9f81ec5cc33",
+)
+
+load("@io_bazel_rules_go//go:def.bzl", "go_register_toolchains", "go_rules_dependencies")
+go_rules_dependencies()
+go_register_toolchains()
+
+# Load the NodeJS rules, check minimum Bazel version, and define the local node_modules.
+RULES_NODEJS_VERSION = "0.11.5"
 http_archive(
     name = "build_bazel_rules_nodejs",
-    url = "https://github.com/bazelbuild/rules_nodejs/archive/0.4.1.zip",
-    strip_prefix = "rules_nodejs-0.4.1",
-    sha256 = "e9bc013417272b17f302dc169ad597f05561bb277451f010043f4da493417607",
+    url = "https://github.com/bazelbuild/rules_nodejs/archive/%s.zip" % RULES_NODEJS_VERSION,
+    strip_prefix = "rules_nodejs-%s" % RULES_NODEJS_VERSION,
+    sha256 = "985bf908faa72cc4638c356a99d19ccac223e5dcd8dae695e3157e5c00f53489",
 )
 
-load("@build_bazel_rules_nodejs//:defs.bzl", "check_bazel_version", "node_repositories")
+load("@build_bazel_rules_nodejs//:package.bzl", "rules_nodejs_dependencies")
+rules_nodejs_dependencies()
```

Comment:
no need to load nodejs dependencies, the rules_typescript_dependencies() will do it

Reply:
Isn't it better to avoid relying on that transitively though? Inside this project we don't know what `rules_typescript` depends on, but we depend directly on `rules_nodejs` for `node_repositories` and `jasmine_node_test` (maybe others too).

Reply:
I'll add a note that we're using it transitively then.